### ISSUE TITLE
FISH_YOY philosophy

### ISSUE THREAD

**Issue by kaijagahm** [MFEh2o/db#70]
Is there a particular reason why FISH_YOY samples aren't included in FISH_SAMPLES? I assumed that the FISH_YOY sampleID's would be a subset of the ones in FISH_SAMPLES, but they seem to be totally separate, making FISH_YOY disconnected from the rest of the database except for projectID and updateID, as far as I can see.

**Comment by kaijagahm** [comment]
@ctsolomon: "OK, these do look like they are all associated with Ziegler YOY mortality survey - the metadataID portion of the sampleID is "YOYSurv.20162017" for all the rows in FISH_YOY. It might be that there's a good reason that these samples can't be accommodated very well in the FISH_SAMPLES structure, but that is a little hard for me to tell at first glance, in part because I don't think that there's a "YOYSurv.20162017" metadata document on Box. @Kaija Gahm, could you please look around for that metadata document, and/or see what else you can learn about the sampling structure by looking at FISH_YOY and at the Ziegler et al. CJFAS paper that reports on these data? I can help with that as well, but won't be until after I am back."

**Comment by kaijagahm** [comment]
Ok, here's what I could dig up. There is a metadata document for that metadataID, it just isn't added to METADATA in the current db version (that will be part of the forthcoming update). The doc is "YOY Largemouth Bass survey 2016&2017 Metadata.docx". The samples were collected on transects, which could be the source of the confusion. In the SITES table, the transects are named e.g. N_5 for the 5th transect at the north site. The fact that there's an underscore in the siteName makes the sampleID's for the yoy samples look all funky: there's an extra underscore where you'd expect the siteID to be a single string. But I don't see any reason why the data needs to be recorded that way; it could just as easily have been N5 instead of N_5, I think (unless maybe there are already sites called e.g. N5 for a different project? Not sure about that).

But the other thing I'm noticing is that FISH_YOY, in terms of the way it's set up and the information it contains, is actually more equivalent to FISH_INFO than to FISH_SAMPLES. Each row is an individual fish, with sampleID's repeated many times. There aren't columns for e.g. date, time, crew, etc. like there are in FISH_SAMPLES. Noticing that initially is why I had expected the sampleID's from FISH_YOY to be uniquely represented in FISH_SAMPLES, to maintain that SAMPLES --> INFO structure. But they aren't there. I also checked SAMPLES to see if they got put there by mistake, but they aren't there either.

So unless I'm missing something, I think that we do not currently have the unique sampleID's from the yoy mortality survey listed anywhere in the database, with their corresponding sampling information. To fix this, we have basically two choices: 1) put them in FISH_SAMPLES and figure out how to deal with the siteID format with the underscore, or 2) create a separate YOY_SAMPLES table. I haven't been able to find a good reason, beyond the siteID format, to do 2), but perhaps there is one that I'm missing.

So, I'm inclined to add these sampleID's to FISH_SAMPLES, provided that we can figure out how to handle the siteID's. But that raises the additional question of whether the current contents of FISH_YOY should be joined with FISH_INFO. This is a philosophy question. Currently, there are several columns in FISH_YOY that aren't in FISH_INFO: YOY (binary) and plus1 (binary, can't figure out from UNITS or the metadata doc what this means), and min/max measurement columns (fish were measured differently in the YOY survey than they are during normal fish sampling, with min/max bins instead of estimated single lengths). However, we've discussed before how the FISH_INFO table is already a behemoth, with a whole bunch of columns that only apply to a few projects, like stomachRemoved, scaleSample, etc. So lumping in the FISH_YOY data and having the additional columns be NA for any non-YOY fish wouldn't be ridiculous, given what we've done in the past. Keeping FISH_YOY separate, on the other hand, kind of raises the question of why e.g. the gonad sample data isn't in its own separate table, and so on and so forth for the other projects relating to fish measurements.

**Comment by kaijagahm** [comment]
Side note: in making a new draft of the db schema for Figshare, I'm realizing that FISH_YOY is totally unconnected from the other db tables (apart from PROJECTS, METADATA, UPDATE_METADATA). The current schema shows it connecting to FISH_SAMPLES (and presumably, to SITES), but it doesn't.

As a stopgap fix, I'm going to add a separate siteID column for this db update.

**Comment by kaijagahm** [comment]
Thanks for weighing in, Colin! That makes a lot of sense.

@joneslabND @ctsolomon I'll bring this issue to our Tuesday meeting so we can think about the different sampling scheme is enough to justify keeping these fish separate, or whether it would be sufficient to lump them into FISH_SAMPLES and make some sort of note that they shouldn't be treated the same way. If so, we should think about where that note would go so that it would be clear enough to users of the database.

If we opt to keep the tables separate, which I also think is a legitimate choice, I'd want to document that in detail somewhere. Possibly in a "tables metadata" doc that I've been thinking about creating, as a sort of high-level introduction to how the database tables relate to each other, what each one is used for, etc.

**Comment by kaijagahm** [comment]
As of 8/25: we've decided to add the sampleID's from FISH_YOY to FISH_SAMPLES, but keep the data in FISH_YOY separate, as an analog to FISH_INFO. This will take care of the different sampling techniques problem, while still preserving the general samples -> data structure.

In order to add these sampleID's, I need to get more information from Jake Ziegler. I've emailed him, as of 8/25. 

Chris notes that we should also go ahead and remove the underscores from the siteID's. In fact, this should already be a rule in the database vocabulary. If it isn't, it should be added.

**Comment by kaijagahm** [comment]
From the information that Jake Ziegler sent me, it looks like there's a whole bunch of info that was recorded in his data sheets but never put into the database. This may end up being a project, and my internet connection is being weird with google docs at the moment, so I'm going to put this aside and come back to it.

**Comment by kaijagahm** [comment]
Next steps here: can go through the data sheets Jake sent to try to make sense of the format (with @Randinotte to help me understand the sites/samples/data format), but then probably need to wait for e.g. crew and time information from the data sheets at cary before we can go farther.

**Comment by kaijagahm** [comment]
> There is a metadata document for that metadataID, it just isn't added to METADATA in the current db version (that will be part of the forthcoming update). The doc is "YOY Largemouth Bass survey 2016&2017 Metadata.docx".

Note that this somehow got added to METADATA twice?? Need to remove.

**Comment by kaijagahm** [comment]
Ok I feel so silly--it turns out that the samples from FISH_YOY actually *were* in FISH_SAMPLES already. They had the wrong metadataID in their sampleID, so I hadn't been able to find them when I searched for them by sampleID. 

I have changed the metadataID portion of the sampleID's in FISH_SAMPLES.

**Comment by kaijagahm** [comment]
I checked to make sure all fish in FISH_YOY were represented in FISH_SAMPLES--they are. I also checked to see how many NFC samples we have (i.e. rows present in FISH_SAMPLES but no associated fish in FISH_YOY). There are a LOT of NFC samples: about 73% of all of the fish_yoy sampleID's are NFC. I ran this by Chris and Jake Ziegler, and they both say it sounds reasonable. That's corroborated by 1) my finding that for sampleID's that **do** have fish, a lot of them have very small numbers of fish, so zero counts seem reasonable, and 2) the graphs in the supplementary material for [Jake's 2019 paper](https://cdnsciencepub.com/doi/abs/10.1139/cjfas-2018-0050) also show a lot of NFC samples.
